plugins/obs-filters/scale-filter.c

```diff
@@ -60,17 +63,27 @@ static void scale_filter_update(void *data, obs_data_t *settings)
 
 	filter->valid = true;
 
-	ret = sscanf(res_str, "%dx%d", &filter->cx_in, &filter->cy_in);
-	if (ret == 2) {
+	if (strcmp(res_str, S_BASE) == 0) {
+		struct obs_video_info ovi;
+		obs_get_video_info(&ovi);
 		filter->aspect_ratio_only = false;
+		filter->base_canvas_resolution = true;
```

Comment:
This needs to be set to false above the strcmp first, to ensure it doesn't get "stuck" as true.